src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php

```diff
@@ -31,13 +31,12 @@ interface ChoiceListFactoryInterface
      * The callable receives the choice as only argument.
      * Null may be passed when the choice list contains the empty value.
      *
-     * @param iterable      $choices The choices
-     * @param callable|null $value   The callable generating the choice
-     *                               values
+     * @param callable|null $value The callable generating the choice
+     *                             values
      *
      * @return ChoiceListInterface The choice list
      */
-    public function createListFromChoices($choices, $value = null);
+    public function createListFromChoices(iterable $choices, $value = null);
```

Comment:
`callable $value = null`?

Reply:
We cannot do this, because then `PropertyAccessDecorator` will not work. It accepts `string|PropertyPath` as well. I will do a PR to fix these phpdocs.

Reply:
`PropertyAccessDecorator` can remove the typehint on `$value` so it accepts more. It should be possible to add the types in the interface as implementations can widen them.